textdb/textdb-exp/src/main/java/edu/uci/ics/textdb/exp/nlp/splitter/NlpSplitOperator.java

```diff
@@ -0,0 +1,181 @@
+    }
+      
+
+    @Override
+    public void open() throws TextDBException {
+        if (cursor != CLOSED) {
+            return;
+        }
+        if (inputOperator == null) {
+            throw new DataFlowException(ErrorMessages.INPUT_OPERATOR_NOT_SPECIFIED);
+        }
+        inputOperator.open();
+        Schema inputSchema = inputOperator.getOutputSchema();
+        
+        // check if input schema is present
+        if (! inputSchema.containsField(predicate.getInputAttributeName())) {
+            throw new RuntimeException(String.format(
+                    "input attribute %s is not in the input schema %s",
+                    predicate.getInputAttributeName(),
+                    inputSchema.getAttributeNames()));
+        }
+        
+        // check if attribute type is valid
+        AttributeType inputAttributeType = 
+                //Does predicate give the correct input attribute name?
+                inputSchema.getAttribute(predicate.getInputAttributeName()).getAttributeType();
+        boolean isValidType = inputAttributeType.equals(AttributeType.STRING) || 
+                inputAttributeType.equals(AttributeType.TEXT);
+        if (! isValidType) {
+            throw new RuntimeException(String.format(
+                    "input attribute %s must have type String or Text, its actual type is %s",
+                    predicate.getInputAttributeName(),
+                    inputAttributeType));
+        }
+        
+        // generate output schema by transforming the input schema
+        outputSchema = transformSchema(inputOperator.getOutputSchema());
+        
+        cursor = OPENED;
+
+    }
+
+    @Override
+    public Tuple getNextTuple() throws TextDBException {
+        if (cursor == CLOSED) {
+            return null;
+        }
+        
+        List<IField> outputFields = new ArrayList<>();
+        
+        if(predicate.getOutputType() == NLPOutputType.ONE_TO_ONE) {
+            currentTuple = inputOperator.getNextTuple();
+            if (currentTuple == null) {
+                return null;
+            }
+            currentSentenceList = getSentenceList(currentTuple);
+            outputFields.addAll(currentTuple.getFields());
+            outputFields.add(new ListField<TextField>(currentSentenceList));
+        }
+        
+        else if (predicate.getOutputType() == NLPOutputType.ONE_TO_MANY) {
+            if(currentSentenceList.isEmpty()) {
+                currentTuple = inputOperator.getNextTuple();
+                if (currentTuple == null) return null;
+                currentSentenceList = getSentenceList(currentTuple);
+            }
+            
+            outputFields.addAll(currentTuple.getFields());
+            //Add the sentences from the current sentence list one by one in the order in which
+            //they were generated in the getSentenceList function
+            outputFields.add(currentSentenceList.remove(0)); //Append a TextField to the output tuple    
```

Comment:
1. The output Schema is now conditionally modified to make it consistent with the output type chosen. 
2. As discussed in the class, I went ahead and added a new field to append to the output schema.